theano/gof/cmodule.py

```diff
@@ -2025,19 +2031,31 @@ def join_options(init_part):
                     _logger.info("g++ -march=native equivalent flags: %s",
                                  GCC_compiler.march_flags)
 
-            # Find working march flag:
-            default_detected_flag = GCC_compiler.march_flags[0][7:]
+            # Find working march flag. If all fail, set march_flags to [] (for x86 archs).
+            default_detected_flag = None
+            march_ind = None
+            mtune_ind = None
+            working_flag = False
+
+            for m_ in xrange(len(GCC_compiler.march_flags)):
+                march_flag = GCC_compiler.march_flags[m_]
+                if 'march' in march_flag:
+                    march_ind = m_
+                    default_detected_flag = march_flag[march_flag.index('=') + 1:]
+                if 'mtune' in GCC_compiler.march_flags[m_]:
+                    mtune_ind = m_
+
             march_flags_to_try = [default_detected_flag, 'corei7-avx', 'corei7', 'core2']
             for march_flag in march_flags_to_try:
-                test_flags = GCC_compiler.march_flags[:]
-                test_flags[0] = '-march=' + march_flag
-                result = try_march_flag(test_flags)
-                if result[0] and result[1]:
-                    GCC_compiler.march_flags[0] = '-march=' + march_flag
-                    GCC_compiler.march_flags[-1] = '-mtune=' + march_flag
+                GCC_compiler.march_flags[march_ind] = '-march=' + march_flag
+                GCC_compiler.march_flags[mtune_ind] = '-mtune=' + march_flag
+                compilation_result, execution_result = try_march_flag(GCC_compiler.march_flags)
+                if compilation_result and execution_result:
+                    working_flag = True
                     break
-            if not result[0] or not result[1]:
-                raise ValueError('Could not find a working march')
+
+            if not working_flag:
+                GCC_compiler.march_flags = []
```

Comment:
If all failed, try:

   for i in ['corei7-avx', 'corei7', 'core2']:
      compilation_result, execution_result = try_march_flag(["-march=" + i])
      ...

it is possible that it is the others flags that pose problems.